src/NServiceBus.Core/Pipeline/Incoming/DeserializeLogicalMessagesConnector.cs

```diff
@@ -7,14 +7,14 @@ namespace NServiceBus
     using System.Linq;
     using System.Threading.Tasks;
     using Logging;
-    using MessageInterfaces.MessageMapper.Reflection;
+    using MessageInterfaces;
     using Pipeline;
     using Transport;
     using Unicast.Messages;
 
     class DeserializeLogicalMessagesConnector : StageConnector<IIncomingPhysicalMessageContext, IIncomingLogicalMessageContext>
     {
-        public DeserializeLogicalMessagesConnector(MessageDeserializerResolver deserializerResolver, LogicalMessageFactory logicalMessageFactory, MessageMetadataRegistry messageMetadataRegistry, MessageMapper mapper)
+        public DeserializeLogicalMessagesConnector(MessageDeserializerResolver deserializerResolver, LogicalMessageFactory logicalMessageFactory, MessageMetadataRegistry messageMetadataRegistry, IMessageMapper mapper)
```

Comment:
Not sure why this change needed to be made. Why use the interface here?

Reply:
Since IMessageMapper is the type available in the settings and I think it’s more to use the interface everywhere (this was the only place). Not against killing it but it would have to be done in a major since it’s public. Does that make sense?